lib/participant_profile_deduplicator.rb

```diff
@@ -48,17 +51,29 @@ def log_overview_info
     log_info("Duplicate profile: #{duplicate_profile_id}")
   end
 
-  def prevent_same_school_different_lead_provider
-    return unless primary_profile.lead_provider&.id != duplicate_profile.lead_provider&.id && primary_profile.school == duplicate_profile.school
+  def reconcile?
+    !same_school_different_lead_provider? || duplicate_has_declarations?
+  end
+
+  def same_school_different_lead_provider?
+    primary_profile.lead_provider&.id != duplicate_profile.lead_provider&.id && school(primary_profile) == school(duplicate_profile)
```

Comment:
I think we might need to do the same here around looking at the latest IR to find the lead provider, similar to schools?